### DISCUSSION TITLE
How to add an icon from gio Resource to gio Menuitem?

### DISCUSSION THREAD

**Discussion by Wanpaku** [gtk-rs/gtk4-rs#1701] [Q&A]
I want to add an icon from gio Resource to gio Menuitem.
Menuitem has a function set_icon() which demands gio::Icon
but I don’t understand how to properly create this gio::Icon from gio::Resource.

**Comment by bilelmoussaoui** [comment]
Use https://gtk-rs.org/gtk-rs-core/stable/latest/docs/gio/struct.ThemedIcon.html#method.new which is a subclass of gio::Icon

**Reply by Wanpaku** [reply]
I did this:
`  let quit_menu_item = gio::MenuItem::new(Some(&quit_title),  Some("app.quit"));

  let quit_icon = gio::ThemedIcon::new("application-exit");

 quit_menu_item.set_icon(&quit_icon);

 let file_menu = gio::Menu::new();

 file_menu.append_item(&quit_menu_item);`

But the icon didn't appeared near "Quit" word in the Menu.
There are only the name of menu item and the hot key to quit app.
